### PR TITLE
Storage mapping validation

### PR DESCRIPTION
**Pull request by jakub-dzon**
Adds storage domain to storage class mapping validation.

Signed-off-by: Jakub Dzon <jdzon@redhat.com>

### CODE REVIEW

File: pkg/providers/ovirt/validation/vm-import-validation.go [hunk 1]
```diff
@@ -159,22 +160,27 @@ func (validator *VirtualMachineImportValidator) processMappingValidationFailures
...
-		return fmt.Errorf("Mapping rules validation failed for %v. Reasons: %s", vmiCrName, message)
+	} else {
+		updateCondition(&copy.Status.Conditions, validationCompletedReason, "Validating completed successfully", true, v2vv1alpha1.Validating)
 	}
 
+	err = validator.client.Status().Update(context.TODO(), copy)
+	if err != nil {
+		return err
+	}
+	if invalid {
```
Comment (pkliczewski) [inline pkg/providers/ovirt/validation/vm-import-validation.go]: Why not to check `invalid` one time and perform `updateCondition` and status update in specific if branch.

Comment (jakub-dzon) [reply pkg/providers/ovirt/validation/vm-import-validation.go]: That would duplicate the code responsible for the update call and its error handling in both branches.

Comment (pkliczewski) [reply pkg/providers/ovirt/validation/vm-import-validation.go]: What about extracting it to a method?

Comment (jakub-dzon) [reply pkg/providers/ovirt/validation/vm-import-validation.go]: Having the update call in both branches will result in exactly the same number of lines as it is now:) The extracted method will only add to it.
I must admit, that when I look at the code changed as you suggested in the first comment, it does look more elegant, despite the duplication. I will change it this way.

**Review by machacekondra — APPROVED** [review]

**Review by pkliczewski — APPROVED** [review]

File: pkg/providers/ovirt/validation/validators/storage-class-provider.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package validators
+
+import (
+	v1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	storage "k8s.io/client-go/kubernetes/typed/storage/v1"
+)
+
+// StorageClasses is responsible for finding storage classes
+type StorageClasses struct {
```
Comment (masayag) [inline pkg/providers/ovirt/validation/validators/storage-class-provider.go:10]: if that the only responsibility of it, perhaps `StorageClassFinder`
if this is expected to grow, no need to rename

Comment (jakub-dzon) [reply pkg/providers/ovirt/validation/validators/storage-class-provider.go:10]: I don't think it will grow in the future - I will rename it.

**Pull request merged**